### PR TITLE
Add action for build, test & lint

### PR DESCRIPTION
**Pull request by rkperes**

### CODE REVIEW

File: .github/workflows/go-foundation.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.20
+
+    - name: Build
+      run: go build -v ./...
```
Comment (cptvictor95) [inline .github/workflows/go-foundation.yml:25]: Later when we have our server and compiler we'll change this to the main server file?

Comment (rkperes) [reply .github/workflows/go-foundation.yml:25]: We should change this to whatever are our "main" files once we have it, yup (and that would be the entrypoint of our server as a binary).
I'm not sure we want to stick to this for building, even... But it'd still make sense for running the tests.
I'd suggest we use a Makefile in the future (although, from my limited knowledge with gh actions, it gets more annoying to set it up to call a make command with all proper dependencies).

Comment (annerland) [reply .github/workflows/go-foundation.yml:25]: Not a problem, we can suffer and figure out this in the future

**Review by annerland — APPROVED** [review]
LGTM

**Pull request merged**